src/algo_keys.c

```diff
@@ -3,40 +3,31 @@
 
 #include "algo_keys.h"
 
-// Allocate 64 bytes for privateKeyData because os_perso_derive_node_bip32
-// appears to write more than 32 bytes to this buffer.  However, we only
-// need 32 bytes for cx_ecfp_init_private_key..
-static uint8_t privateKeyData[64];
-
-uint8_t publicKey[32];
 
 void
-algorand_key_derive(void)
+algorand_key_derive(uint32_t accountId, cx_ecfp_private_key_t *privateKey)
 {
-  uint32_t bip32Path[5];
+  static uint8_t  privateKeyData[64];
+  static uint32_t bip32Path[5];
 
   bip32Path[0] = 44  | 0x80000000;
   bip32Path[1] = 283 | 0x80000000;
-  bip32Path[2] = 0   | 0x80000000;
+  bip32Path[2] = accountId | 0x80000000;
```

Comment:
No check needed on `accountId` value? (idk algorand but I was wondering if any value in the 31-bit range is allowed as an account i.e. if the protocol is defining some limitations that could be checked here?)

Reply:
Any 31bit value is valid :+1: